sdk/couchbase-core/src/queryx/query.rs

```diff
@@ -138,4 +147,538 @@ impl<C: Client> Query<C> {
 
         QueryRespReader::new(res, &self.endpoint, statement, client_context_id).await
     }
+
+    pub async fn get_all_indexes(&self, opts: &GetAllIndexesOptions) -> error::Result<Vec<Index>> {
+        let mut where_clause = if opts.collection_name.is_none() && opts.scope_name.is_none() {
+            if !opts.bucket_name.is_empty() {
+                let encoded_bucket = encode_value(&opts.bucket_name)?;
+                format!(
+                    "(keyspace_id={} AND bucket_id IS MISSING) OR bucket_id={}",
+                    encoded_bucket, encoded_bucket
+                )
+            } else {
+                "1=1".to_string()
+            }
+        } else if opts.collection_name.is_some() && opts.scope_name.is_some() {
+            let scope_name = normalise_default_name(opts.scope_name.as_ref().unwrap());
```

Comment:
I feel like there must be a more idiomatic way of doing this than checking is_some and then unwrapping.

Reply:
Nice!